data/mods/mixandmega/scripts.ts

```diff
@@ -8,6 +8,390 @@ export const Scripts: ModdedBattleScriptsData = {
 			this.modData('FormatsData', id).isNonstandard = null;
 		}
 	},
+	start() {
+		// Deserialized games should use restart()
+		if (this.deserialized) return;
+		// need all players to start
+		if (!this.sides.every(side => !!side)) throw new Error(`Missing sides: ${this.sides}`);
+
+		if (this.started) throw new Error(`Battle already started`);
+
+		const format = this.format;
+		this.started = true;
+		if (this.gameType === 'multi') {
+			this.sides[1].foe = this.sides[2]!;
+			this.sides[0].foe = this.sides[3]!;
+			this.sides[2]!.foe = this.sides[1];
+			this.sides[3]!.foe = this.sides[0];
+			this.sides[1].allySide = this.sides[3]!;
+			this.sides[0].allySide = this.sides[2]!;
+			this.sides[2]!.allySide = this.sides[0];
+			this.sides[3]!.allySide = this.sides[1];
+			// sync side conditions
+			this.sides[2]!.sideConditions = this.sides[0].sideConditions;
+			this.sides[3]!.sideConditions = this.sides[1].sideConditions;
+		} else {
+			this.sides[1].foe = this.sides[0];
+			this.sides[0].foe = this.sides[1];
+			if (this.sides.length > 2) { // ffa
+				this.sides[2]!.foe = this.sides[3]!;
+				this.sides[3]!.foe = this.sides[2]!;
+			}
+		}
+
+		for (const side of this.sides) {
+			this.add('teamsize', side.id, side.pokemon.length);
+		}
+
+		this.add('gen', this.gen);
+
+		this.add('tier', format.name);
+		if (this.rated) {
+			if (this.rated === 'Rated battle') this.rated = true;
+			this.add('rated', typeof this.rated === 'string' ? this.rated : '');
+		}
+
+		if (format.onBegin) format.onBegin.call(this);
+		for (const rule of this.ruleTable.keys()) {
+			if ('+*-!'.includes(rule.charAt(0))) continue;
+			const subFormat = this.dex.formats.get(rule);
+			if (subFormat.onBegin) subFormat.onBegin.call(this);
+		}
+		for (const pokemon of this.getAllPokemon()) {
+			const item = pokemon.getItem();
+			if (['adamantcrystal', 'griseouscore', 'lustrousglobe', 'vilevial'].includes(item.id) &&
+				item.forcedForme !== pokemon.species.name) {
+				// @ts-ignore
+				const rawSpecies = this.actions.getMixedSpecies(pokemon.m.originalSpecies, item.forcedForme!, pokemon);
+				const species = pokemon.setSpecies(rawSpecies);
+				if (!species) continue;
+				pokemon.baseSpecies = rawSpecies;
+				pokemon.details = species.name + (pokemon.level === 100 ? '' : ', L' + pokemon.level) +
+					(pokemon.gender === '' ? '' : ', ' + pokemon.gender) + (pokemon.set.shiny ? ', shiny' : '');
+				pokemon.setAbility(species.abilities['0'], null, true);
+				pokemon.baseAbility = pokemon.ability;
+			}
+		}
```

Comment:
Okay, it looks like you're overriding `start` to put this here? Can this be moved to `onBegin`? It's very brittle to maintain a separate copy of the `start` script.